File: backend/planta-smart-4.0/src/main/java/com/smart/appsa/model/enums/StatusEstacao.java
package com.smart.appsa.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

import jakarta.persistence.EnumeratedValue;

public enum StatusEstacao {
    PARADO(0),
    OCUPADO(1),
    AGUARDANDO(2),
    MANUAL(3),
    EMERGENCIA(4);

    @EnumeratedValue
    int value;

    @JsonCreator
    public static StatusEstacao fromValue(String value) {
        Integer parsed = Integer.valueOf(value);
        for (StatusEstacao s : values()) {
            if (parsed.equals(s.value)) return s;
        }
        throw new IllegalArgumentException("StatusEstacao inválido: " + value);
    }

    private StatusEstacao(Integer value) {
        this.value = value;
    }

    public Integer getValue() { return value; }
}

File: frontend/planta-smart-4.0/src/domain/enums/StatusEstacao.ts
export enum StatusEstacao {
  Parado = 'parado',
  Ocupado = 'ocupado',
  Aguardando = 'aguardando',
  Manual = 'manual',
  Emergencia = 'emergencia',
}

const statusEstacaoMap: Record<number, StatusEstacao> = {
  0: StatusEstacao.Parado,
  1: StatusEstacao.Ocupado,
  2: StatusEstacao.Aguardando,
  3: StatusEstacao.Manual,
  4: StatusEstacao.Emergencia,
};

const statusEstacaoStringMap: Record<string, StatusEstacao> = {
  PARADO: StatusEstacao.Parado,
  OCUPADO: StatusEstacao.Ocupado,
  AGUARDANDO: StatusEstacao.Aguardando,
  MANUAL: StatusEstacao.Manual,
  EMERGENCIA: StatusEstacao.Emergencia,
};

export namespace StatusEstacao {
  export function fromValue(value: number): StatusEstacao {
    const status = statusEstacaoMap[value];
    if (status === undefined) {
      throw new Error(`StatusEstacao inválido recebido do backend: ${value}`);
    }
    return status;
  }

  export function fromString(value: string): StatusEstacao {
    const status = statusEstacaoStringMap[value.toUpperCase()];
    if (status === undefined) {
      throw new Error(`StatusEstacao inválido recebido do backend: "${value}"`);
    }
    return status;
  }
}